c7n/resources/account.py

```diff
@@ -1574,7 +1574,7 @@ def get_target(self, resource):
         resource[self.annotation].update(encryption_setting.get('ConnectionPasswordEncryption'))
 
         for kmskey in self.data:
-            if not self.data[kmskey].startswith('alias'):
+            if not isinstance(self.data[kmskey], str) or not self.data[kmskey].startswith('alias'):
```

Comment:
given how simple this schema is, we really shouldn't be using shape validate on it.

shape validation is a trade off, not a pancea.. it leads to poor docs and poor editor integration when using the jsonschema. it should really only be used for complex pass through schemas, else its much nicer to use jsonschema for end users.

Reply:
its also now apparent that this wasn't even picking the right shape to begin with, it should have been using EncryptionAtRest shape.. so unfortunately this added an extraneous unneeded key for all users to specify :-(

Reply:
Hello @kapilt ! Thanks for looking into it! I don't see any shape validation happening here. Maybe i'm missing something. I see there's a schema specified here
```    schema = {
        'type': 'object',
        'additionalProperties': False,
        'properties': {
            'type': {'enum': ['glue-security-config']},
            'CatalogEncryptionMode': {'type': 'string'},
            'SseAwsKmsKeyId': {'type': 'string'},
            'ReturnConnectionPasswordEncrypted': {'type': 'boolean'},
            'AwsKmsKeyId': {'type': 'string'}
        }
    }
```

Reply:
see the glue catalog set-encryption action, re shape

Reply:
so the two set-encryption actions are completely inconsistent, one using gratuitous shape validation, and the other using schema with completely different syntax and fields. thats a very unfortunate oversight, given its been out in the world less then two weeks, I'm inclined to fix the mess now before it spreads, even if it means breaking compatibility.

Reply:
Got you @kapilt ! Let me know what do you feel is the best path moving forward. I can help in whatever way I can

Reply:
thanks, I'm just gonna convert it the shape validation to the jsonschema, and ditch the validate method here.

Reply:
I was thinking as per usage compatibility we could leverage the inbuilt value filter for `resource: glue-catalog` to check the encryption states and `actions: set-encryption` to set them. Since this additional filter `glue-security-config` was already present on accounts I just extended it to `glue-catalog`